main.cpp

```diff
@@ -42,7 +42,139 @@ send me a DM to check your pull request
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs );
+    float subtract( float lhs, float rhs );
+    float multiply( float lhs, float rhs );
+    float divide( float lhs, float rhs );
+};
+
+float FloatType::add( float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract( float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply( float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide( float lhs, float rhs)
+{
+    if (rhs == 0 )
```

Comment:
You need to do an `epsilon` comparison here to suppress the warning.
you basically compare if the absolute difference between your `rhs` and some other number is smaller than some epsilon, then that means your `rhs` is basically 0.
- [ ] revise and add epsilon comparison.